tools/ld-analyse/CMakeLists.txt

```diff
@@ -0,0 +1,246 @@
+# DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+# (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+# LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+# ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+# SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+# The views and conclusions contained in the software and documentation are those
+# of the authors and should not be interpreted as representing official policies, 
+# either expressed or implied, of the FreeBSD Project.
+#=============================================================================
+
+if (Qt5Gui_FOUND)
+  get_target_property(QT_INCLUDE_DIR Qt5::Gui INTERFACE_INCLUDE_DIRECTORIES)
+endif ()
+
+find_path ( QWT_INCLUDE_DIR
+  NAMES qwt_plot.h
+  HINTS ${QT_INCLUDE_DIR}
+  PATH_SUFFIXES qwt qwt-qt5 qwt6
+)
+
+set ( QWT_INCLUDE_DIRS ${QWT_INCLUDE_DIR} )
+
+# version
+set ( _VERSION_FILE ${QWT_INCLUDE_DIR}/qwt_global.h )
+if ( EXISTS ${_VERSION_FILE} )
+  file ( STRINGS ${_VERSION_FILE} _VERSION_LINE REGEX "define[ ]+QWT_VERSION_STR" )
+  if ( _VERSION_LINE )
+    string ( REGEX REPLACE ".*define[ ]+QWT_VERSION_STR[ ]+\"([^\"]*)\".*" "\\1" QWT_VERSION_STRING "${_VERSION_LINE}" )
+  endif ()
+endif ()
+unset ( _VERSION_FILE )
+
+find_library ( QWT_LIBRARY
+  NAMES qwt qwt-qt5
+  HINTS ${QT_LIBRARY_DIR}
+)
+
+set ( QWT_LIBRARIES ${QWT_LIBRARY} )
+
+include ( FindPackageHandleStandardArgs )
+find_package_handle_standard_args( Qwt REQUIRED_VARS QWT_LIBRARY QWT_INCLUDE_DIR VERSION_VAR QWT_VERSION_STRING )
+
+if (Qwt_FOUND AND NOT TARGET Qwt::Qwt)
+  add_library(Qwt::Qwt UNKNOWN IMPORTED)
+  set_target_properties(Qwt::Qwt PROPERTIES
+                        INTERFACE_INCLUDE_DIRECTORIES "${QWT_INCLUDE_DIRS}"
+                        IMPORTED_LOCATION "${QWT_LIBRARIES}")
+endif ()
+
+mark_as_advanced (
+  QWT_LIBRARY
+  QWT_INCLUDE_DIR
+)
+
+#
+#
+#
+#
+#
+#
+#
+#
+# Things above this line can go to an external file if possible
+# Anything beneath stays here
+#
+#
+#
+#
+#
+#
+#
+#
+
+cmake_minimum_required(VERSION 3.22)
+project(ld-analyse)
+
+set(CMAKE_INCLUDE_CURRENT_DIR ON)
+#list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}/cmake_modules")
+
+# Set up AUTOMOC and some sensible defaults for runtime executionGui
+# When using Qt 6.3, you can replace the code block below with
+#qt_standard_project_setup()
+set(CMAKE_AUTOMOC ON)
+include(GNUInstallDirs)
+set(CMAKE_AUTOUIC ON)
+
+find_library(QWT_LIBRARY_RELEASE NAMES qwt qwt-qt5 qt5-qwt PATH_SUFFIXES lib)
+find_package(QT NAMES Qt5 REQUIRED COMPONENTS Core Gui Widgets HINTS ${QT_LIBRARY_DIR})
+find_package(Qt${QT_VERSION_MAJOR} REQUIRED COMPONENTS Core Gui Widgets)
+
+# Name all the files
+add_executable(ld-analyse MACOSX_BUNDLE
```

Comment:
Is it necessary to specify the .h files here? (It may well be for Qt's autogeneration stuff to work - it's just that normally CMake can figure out the headers for itself.)

Reply:
I based this file off of [the top-level CMakeLists.txt file of DomesdayDuplicator](https://github.com/oyvindln/DomesdayDuplicator/blob/master/Linux-Application/DomesdayDuplicator/CMakeLists.txt), which is also designed for GUIs